torch/autocast_mode.py

```diff
@@ -161,22 +179,28 @@ def __init__(self, device_type, enabled=True, **kwargs):
         self._enabled = enabled
 
     def __enter__(self):
+        if torch._jit_internal.is_scripting():
+            return self
```

Comment:
Let me double check what graph would look like. I think technically, we would ***only*** want to set the global values at __enter__, but not __init__. I don't know if there's any issue with that and eventually drive the decision to put it in __init__ instead.

Reply:
yea, we would want to do it in enter and exit.. setting the global values would be nice because it would remove the restriction on needing to see the complete graph, for example if there was an interface call (although thats pretty much a guarded feature).